assets/js/components/surveys/SurveyShow.jsx

```diff
@@ -89,7 +93,22 @@ class SurveyShow extends Component<any, State> {
 
   showHistograms() {
     const { survey } = this.props
-    return survey && survey.state == "running"
+    return survey && (survey.state == "running" || survey.state == "paused")
+  }
+
+  pauseOrResumeSurvey() {
+    const { dispatch, survey, projectId, surveyId } = this.props
+    if (survey && survey.state == "running") {
+      pauseSurvey(projectId, surveyId).then(() => {
+        dispatch(actions.pauseSurvey())
+        this.setState({ paused: true })
```

Comment:
Definitely have to ditch the `paused` variable in the state and make it a function of the survey's state. You don't want to have to remember to update the `this.setState` every time you invoke an action that changes the survey state.